src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -241,8 +243,26 @@ QSharedPointer<Database> CsvImportWidget::buildDatabase()
     db->rootGroup()->setNotes(tr("Imported from CSV file: %1").arg(m_filename));
 
     auto rows = m_parserModel->rowCount() - m_parserModel->skippedRows();
+
+    // Check for common root group
+    QString rootGroupName;
+    for (int r = 0; r < rows; ++r) {
+        auto groupPath = m_parserModel->data(m_parserModel->index(r, 0)).toString();
+        auto groupName = groupPath.mid(0, groupPath.indexOf('/'));
+        if (!rootGroupName.isNull() && rootGroupName != groupName) {
+            rootGroupName.clear();
+            break;
+        }
+        rootGroupName = groupName;
+    }
+
+    if (!rootGroupName.isEmpty()) {
+        db->rootGroup()->setName(rootGroupName);
+    }
```

Comment:
I don't understand what the intent of this code is. Assuming all or some entries have different first path components, this will just take the last entry's root as overall root group, which seems wrong to me.

I suggest we check that all entries have the same root as the first one and if there are no mismatches, we use that. Otherwise we insert a new root group on top.